### PR TITLE
Acl delete and revert bots

### PR DESCRIPTION
**Pull request by Alin-Loshevsky**

### CODE REVIEW

File: bots/acl_delete.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        return f'Unexpected error: {e}\n'
+
+
+def run_action(boto_session, rule, entity, params):
+
+    text_output = ''
+    ec2_resource = boto_session.resource('ec2')
+    try:
+        # getting the network acl from cloud trail event
+        acl_description = find_event(boto_session, entity)
```
Comment (RavidAtias) [inline bots/acl_delete.py:39]: I think there is util func for it

File: bots/acl_delete.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        acl_id = acl_description.get('networkAclId')
```
Comment (RavidAtias) [inline bots/acl_delete.py:41]: is it acl_id or description? name should be clear

File: bots/acl_delete.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        network_acl = ec2_resource.NetworkAcl(acl_id)
+        # checking if acl was not creates as default vpc's network acl
+        if not network_acl.is_default:
```
Comment (RavidAtias) [inline bots/acl_delete.py:45]: If you do not delete default you need to add it to the documentation as limitation

File: bots/acl_revert_modification.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+    return cloud_trail.get('eventName'), event_params
+
+
+def run_action(boto_session, rule, entity, params):
+    text_output = ''
+    acl_id = entity.get('id')
+
+    # There are 3 possible events. Each has it's own solution. This is a dict to help assign a function to each case.
+    functions = {'ReplaceNetworkAclEntry': replace_entry, 'DeleteNetworkAclEntry': create_entry, 'CreateNetworkAclEntry': delete_entry}
```
Comment (RavidAtias) [inline bots/acl_revert_modification.py:179]: give better name - something like event_name_to_function_mapping

File: bots/acl_revert_modification.py [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+    except ClientError as e:
+
+        text_output = f'Unexpected error: {e}\n'
+
+
+    return text_output
+
+
+def find_event(boto_session, attribute_value, entity={}):
```
Comment (RavidAtias) [inline bots/acl_revert_modification.py:128]: Should be as util - and why there are prints in here?

File: bots/acl_revert_modification.py [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+
+def delete_entry(boto_session, params):
+    '''
+    function will be called if event is: CreateNetworkAclEntry
+    it will delete the new created entry.
+    '''
+
+    text_output = ''
+    ec2_resource = boto_session.resource('ec2')
```
Comment (RavidAtias) [inline bots/acl_revert_modification.py:110]: try to use boto_session.client

File: bots/acl_revert_modification.py [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+    acl_id = params.get('networkAclId')
+    # Trying to get entry creation event
+    event_name, event_params = find_event(boto_session, acl_id)
+
+    # Exiting if creation event can not be found
+    if event_name != 'CreateNetworkAclEntry':
+        text_output = f'Can not recreate deleted rule {params.get("ruleNumber")}.\nExiting'
+        return text_output
+
+    ec2_resource = boto_session.resource('ec2')
```
Comment (RavidAtias) [inline bots/acl_revert_modification.py:79]: try to use boto_session.client

File: bots/acl_revert_modification.py [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+    In this function params are not used in function create_entry because they lack information needed to create an entry.
+    '''
+
+    text_output = ''
+    acl_id = params.get('networkAclId')
+    # Trying to get entry creation event
+    event_name, event_params = find_event(boto_session, acl_id)
+
+    # Exiting if creation event can not be found
+    if event_name != 'CreateNetworkAclEntry':
```
Comment (RavidAtias) [inline bots/acl_revert_modification.py:75]: String as this one should be const

File: bots/acl_revert_modification.py [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+    '''
+
+    text_output = ''
+    acl_id = params.get('networkAclId')
+    # Trying to get entry creation event
+    print('here')
+    event_name, event_params = find_event(boto_session, acl_id)
+
+    # Exiting if creation event can not be found
+    if event_name != 'CreateNetworkAclEntry':
```
Comment (RavidAtias) [inline bots/acl_revert_modification.py:32]: const

**Review by RavidAtias — CHANGES_REQUESTED** [review]

File: bots/acl_delete.py (hunk 1 above)
Comment (Alin-Lo) [reply bots/acl_delete.py:39]: I use the util func and extract the acl from it's response

File: bots/acl_delete.py (hunk 2 above)
Comment (Alin-Lo) [reply bots/acl_delete.py:41]: It is an id that I get from description

File: bots/acl_revert_modification.py (hunk 5 above)
Comment (Alin-Lo) [reply bots/acl_revert_modification.py:128]: I need to use this function in two different situations. one time I use the utils func and other times when I dont have a time, I use look up and check all events related to the acl.

File: bots/acl_delete.py (hunk 1 above)
Comment (RavidAtias) [reply bots/acl_delete.py:39]: So give proper name such as find_event_and_get_acl

File: bots/acl_delete.py [hunk 10]
```diff
@@ -0,0 +1,52 @@
+'''
+## acl_delete
+What it does: deletes created network acl.
+Usage: AUTO: acl_delete
+
+Sample GSL: cloudtrail where event.name='CreateNetworkAcl'
+Limitation: Function will not delete default vpc's network acl
```
Comment (RavidAtias) [inline bots/acl_delete.py:7]: Change function to bot

File: bots/acl_revert_modification.py (hunk 5 above)
Comment (RavidAtias) [reply bots/acl_revert_modification.py:130]: @doronzil What do you think about this function ?

**Review by RavidAtias — CHANGES_REQUESTED** [review]

File: bots/acl_revert_modification.py (hunk 5 above)
Comment (doronzil) [reply bots/acl_revert_modification.py:130]: You can change "cloudtrail_event_lookup" function in utils to support "no time" lookups. please edit it and use it :)

File: bots/acl_delete.py [hunk 11]
```diff
@@ -0,0 +1,53 @@
+'''
+## acl_delete
+What it does: deletes created network acl.
+Usage: AUTO: acl_delete
+
+Sample GSL: cloudtrail where event.name='CreateNetworkAcl'
+Limitation: Bot will not delete default vpc's network acl
+Note: Logic only bot
```
Comment (Alin-Lo) [inline bots/acl_delete.py]: should this note be here? 
Is compliance alert provides acl id in entity when calling the bot? 
if not the note is relevant.
@RavidAtias

Comment (doronzil) [reply bots/acl_delete.py]: Is there an option, ACL is returned as an entity in Compliance? @RavidAtias

File: bots/acl_delete.py [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+        return cloud_trail.get("responseElements")['networkAcl']
+
+    except ClientError as e:
+        return f'Unexpected error: {e}\n'
+
+
+def run_action(boto_session, rule, entity, params):
+
+    text_output = ''
+    ec2_resource = boto_session.resource('ec2')
```
Comment (doronzil) [inline bots/acl_delete.py:37]: use boto3 client

**Review by doronzil — APPROVED** [review]

**Review by RavidAtias — APPROVED** [review]

**Pull request merged**